test/unit/api/op_black.cpp

```diff
@@ -12,186 +12,169 @@
  ** \brief Black box testing of the Term class
  **/
 
-#include <cxxtest/TestSuite.h>
-
-#include "api/cvc4cpp.h"
+#include "test_api.h"
 
 using namespace CVC4::api;
 
-class OpBlack : public CxxTest::TestSuite
+class TestApiOpBlack : public TestApi
 {
- public:
-  void setUp() override {}
-  void tearDown() override {}
-
-  void testGetKind();
-  void testIsNull();
-  void testOpFromKind();
-  void testGetIndicesString();
-  void testGetIndicesUint();
-  void testGetIndicesPairUint();
-
-  void testOpScopingToString();
-
- private:
-  Solver d_solver;
 };
 
-void OpBlack::testGetKind()
+TEST_F(TestApiOpBlack, getKind)
 {
   Op x;
   x = d_solver.mkOp(BITVECTOR_EXTRACT, 31, 1);
-  TS_ASSERT_THROWS_NOTHING(x.getKind());
+  ASSERT_NO_THROW(x.getKind());
 }
 
-void OpBlack::testIsNull()
+TEST_F(TestApiOpBlack, isNull)
 {
   Op x;
-  TS_ASSERT(x.isNull());
+  ASSERT_TRUE(x.isNull());
   x = d_solver.mkOp(BITVECTOR_EXTRACT, 31, 1);
-  TS_ASSERT(!x.isNull());
+  ASSERT_FALSE(x.isNull());
 }
 
-void OpBlack::testOpFromKind()
+TEST_F(TestApiOpBlack, opFromKind)
 {
   Op plus(&d_solver, PLUS);
-  TS_ASSERT(!plus.isIndexed());
-  TS_ASSERT_THROWS(plus.getIndices<uint32_t>(), CVC4ApiException&);
+  ASSERT_FALSE(plus.isIndexed());
+  ASSERT_THROW(plus.getIndices<uint32_t>(), CVC4ApiException);
 
-  TS_ASSERT_THROWS_NOTHING(d_solver.mkOp(PLUS));
-  TS_ASSERT_EQUALS(plus, d_solver.mkOp(PLUS));
-  TS_ASSERT_THROWS(d_solver.mkOp(BITVECTOR_EXTRACT), CVC4ApiException&);
+  ASSERT_NO_THROW(d_solver.mkOp(PLUS));
+  EXPECT_EQ(plus, d_solver.mkOp(PLUS));
+  ASSERT_THROW(d_solver.mkOp(BITVECTOR_EXTRACT), CVC4ApiException);
 }
 
-void OpBlack::testGetIndicesString()
+TEST_F(TestApiOpBlack, getIndicesString)
 {
   Op x;
-  TS_ASSERT_THROWS(x.getIndices<std::string>(), CVC4ApiException&);
+  ASSERT_THROW(x.getIndices<std::string>(), CVC4ApiException);
```

Comment:
One general comment. Do you want to continue the rest of the test even if this fails? In general, it's recommended to use the `EXPECT_*` macros unless it really doesn't make sense to continue the test.

Reply:
I did consider which ones to convert to `EXPECT` and decided to have all the `THROW` and `NO_THROW` as `ASSERT`, that is, not continuing. I guess we could discuss this rule of thumb, it's not set in stone. Past experience has shown me though that (at least for me) it doesn't make much of a difference if it continues or not -- I usually fix the first issue anyways and then continue. Note that `ASSERT` failures only terminate the current test (as in the current function that is executed), not the whole unit test module (as in, execution of the unit test binary, e.g., op_black).

Reply:
I think that rule of thumb makes sense! What about the `ASSERT_TRUE`s? I don't feel strongly about this by the way, just wanted to think about it up front because I ended up using `ASSERT` in smt-switch and now am slowly changing things to `EXPECT`. You make a good point about it not making much of a difference though, because you usually need to fix everything eventually anyway. I also tend to rebuild/rerun the test after every fix.

Reply:
There I tried to use `EXPECT` if it's independent from what comes after. For example, if you just check a value or something. But if you assert the size of, let's say, a vector, and then access the elements assuming the size (e.g., we assert that size is 2 and then access with `foo[0]` `foo[1]`) I used assert, because if that assertion fails, follow-up statements might crash.